src/SignalR/common/Shared/ReflectionHelper.cs

```diff
@@ -13,6 +15,13 @@ internal static class ReflectionHelper
         public static bool IsStreamingType(Type type, bool mustBeDirectType = false)
         {
             // TODO #2594 - add Streams here, to make sending files easy
+
+#if NETCOREAPP3_0
+            if (IsIAsyncEnumerable(type))
```

Comment:
Remind me, what is this for?

Reply:
For distinguishing streaming params vs normal params.

Reply:
Exactly . This is just the helper specifically for detecting IAsyncEnumerable